tests/python_tests/test_vlm_pipeline.py

```diff
@@ -133,131 +283,101 @@ def implementation():
-    handwritten_url = "https://github.com/user-attachments/assets/8c9ae017-7837-4abc-ae92-c1054c9ec350"
-    return openvino.Tensor(from_cache_or_download(pytestconfig, handwritten_url, "handwritten.png"))
+def cat_image_32x32(cat_image):
+    return cat_image.resize((32, 32))
 
 
-model_ids = [
-    "katuni4ka/tiny-random-minicpmv-2_6",
-    "katuni4ka/tiny-random-phi3-vision",
-    "katuni4ka/tiny-random-phi-4-multimodal",
-    "katuni4ka/tiny-random-llava",
-    "katuni4ka/tiny-random-llava-next",
-    "katuni4ka/tiny-random-internvl2",
-    "katuni4ka/tiny-random-qwen2vl",
-    "katuni4ka/tiny-random-qwen2.5-vl",
-    "katuni4ka/tiny-random-gemma3",
-    "qnguyen3/nanoLLaVA"
-]
+@pytest.fixture(scope="module")
+def cat_tensor(cat_image) -> openvino.Tensor:
+    return openvino.Tensor(cat_image)
 
-# On macOS, transformers<4.52 is required, but this causes gemma3 to fail
-GEMMA3_MACOS_XFAIL_REASON = "gemma3 not supported on macOS with older transformers"
 
-if sys.platform == "darwin":
-    model_ids = [
-        pytest.param(
-            model_id,
-            marks=pytest.mark.xfail(reason=GEMMA3_MACOS_XFAIL_REASON)
-        ) if "gemma3" in model_id else model_id
-        for model_id in model_ids
-    ]
+@pytest.fixture(scope="module")
+def car_tensor(pytestconfig: pytest.Config) -> openvino.Tensor:
+    return openvino.Tensor(from_cache_or_download(pytestconfig, TEST_IMAGE_URLS['car'], "car.jpg"))
 
-attention_backend = ["PA", "SDPA"]
+
+@pytest.fixture(scope="module")
+def handwritten_tensor(pytestconfig: pytest.Config) -> openvino.Tensor:
+    return openvino.Tensor(from_cache_or_download(pytestconfig, TEST_IMAGE_URLS['handwritten'], "handwritten.png"))
 
 
 @pytest.mark.precommit
-@pytest.mark.parametrize("model_id", model_ids)
-@pytest.mark.parametrize("backend", attention_backend)
-def test_vlm_pipeline(model_id, backend, cat_tensor, handwritten_tensor, car_tensor):
+@parametrize_all_models
+def test_vlm_pipeline(ov_pipe_model: VlmModelInfo, cat_tensor, handwritten_tensor, car_tensor):
+    ov_pipe = ov_pipe_model.pipeline
     def streamer(word: str) -> bool:
         nonlocal result_from_streamer
         result_from_streamer.append(word)
         return False
 
-    models_path = get_ov_model(model_id)
-    ov_pipe = VLMPipeline(models_path, "CPU", ATTENTION_BACKEND=backend)
-    generation_config = ov_pipe.get_generation_config()
-    generation_config.max_new_tokens = 30
-    generation_config.set_eos_token_id(ov_pipe.get_tokenizer().get_eos_token_id())
+    generation_config = _setup_generation_config(ov_pipe)
 
     for images in [], [cat_tensor], [cat_tensor, handwritten_tensor, car_tensor]:
         result_from_streamer = []
         res = ov_pipe.generate(
-            prompts[0],
+            PROMPTS[0],
             images=images,
             generation_config=generation_config,
             streamer=streamer,
         )
         assert res.texts[0] == "".join(result_from_streamer)
```

Comment:
Looks like "three <del>raccoons</del> tests stacked on top of each other in a trench coat pretending to be <del>a person</del> one test"
It is better to create a fixture that will yield `[], [cat_tensor], [cat_tensor, handwritten_tensor, car_tensor]` one by one.
`PROMPTS[0]` should be a parameter as well.